src/grt/src/GlobalRouter.tcl

```diff
@@ -277,12 +277,26 @@ proc repair_antennas { args } {
       # repairAntennas locates diode
       set diode_mterm "NULL"
     } elseif { [llength $args] == 1 } {
-      set diode_port_name [lindex $args 0]
-      set diode_port [sta::get_lib_pins -quiet $diode_port_name]
-      if { $diode_port == "" } {
-        utl::error GRT 69 "Diode $diode_port_name not found."
+      set db [ord::get_db]
+      set diode_cell [lindex $args 0]
+
+      set diode_master [$db findMaster $diode_cell]
+      if { $diode_master == "NULL" } {
+        utl::error GRT 69 "Diode cell $diode_cell not found."
+      }
+      
+      set diode_mterms [$diode_master getMTerms]
+      set non_pg_count 0
+      foreach mterm $diode_mterms {
+        if { [$mterm getSigType] != "POWER" && [$mterm getSigType] != "GROUND" } {
+          set diode_mterm $mterm
+          incr non_pg_count
+        }
+      }
+
+      if { $non_pg_count > 1 } {
+        utl::warn "Diode cell has more than one non-pg port."
```

Comment:
I think it should be an error since we don't know which mterm to use in this case.